### ISSUE TITLE
Return null value after few Days

### ISSUE THREAD

**Issue by Ruwasoft** [ivehement/saf#17]
I used this package to read the files inside folder. Its working fine. But after few days, Its returning null value.

**Comment by JasminScalelot** [comment]
I have the same problem, Every time need to request for getDirectoryPermission other wise it's return null file list.

**Comment by AMuellerAtAHS** [comment]
Maybe this is a limitation of platform specific Storage Access Framework.
The thing is, that the user should always grant the permission, when the app needs it. So that the app couldn't do any mystical stuff in the background.

**Comment by yashLadha** [comment]
@AMuellerAtAHS is correct, it might be due to permission denial error in log of the application. We might need a way to refresh the permission elegantly. I am searching if there is a workaround.

**Comment by HussnulMaab192** [comment]
@yashLadha > 
have you found any solution?
I am facing same problem!

**Comment by AnkitFlutterDev** [comment]
@AMuellerAtAHS i dont think this is a limitation of platform specific Storage Access Framework,cuz i have tried shared_storage plugin in my demo project,and there no need to allow permission every time even with Restart.


Please clarify this.
https://pub.dev/packages/shared_storage

**Comment by jvoltci** [comment]
2.0.0 is a major rewrite (new URI-based API, fixed external-volume and permission handling, native read/write). This is likely resolved — could you retry with `saf: ^2.0.0` and reopen with details if it still happens? Docs: https://jvoltci.github.io/saf/

**Comment by jvoltci** [comment]
Closing: addressed by the 2.0.0 URI-based rewrite. Please reopen with a repro on 2.0 if it still occurs.
